modules/vmss/main.tf

```diff
@@ -1,259 +1,370 @@
+locals {
+  operator = {
+    ">"  = "GreaterThan"
+    ">=" = "GreaterThanOrEqual"
+    "<"  = "LessThan"
+    "<=" = "LessThanOrEqual"
+    "==" = "Equals"
+    "!=" = "NotEquals"
+  }
+
+  panos_metrics = [
+    "DataPlanePacketBufferUtilization",
+    "panSessionThroughputPps",
+    "panSessionThroughputKbps",
+    "panSessionActive",
+    "panSessionUtilization",
+    "DataPlaneCPUUtilizationPct"
+  ]
+
+  password = sensitive(var.authentication.password)
+}
+
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_virtual_machine_scale_set
 resource "azurerm_linux_virtual_machine_scale_set" "this" {
-  name                            = var.name
-  location                        = var.location
-  resource_group_name             = var.resource_group_name
-  admin_username                  = var.username
-  admin_password                  = var.disable_password_authentication ? null : var.password
-  disable_password_authentication = var.disable_password_authentication
-  encryption_at_host_enabled      = var.encryption_at_host_enabled
-  overprovision                   = var.overprovision
-  platform_fault_domain_count     = var.platform_fault_domain_count
-  proximity_placement_group_id    = var.proximity_placement_group_id
-  single_placement_group          = var.single_placement_group
-  instances                       = var.autoscale_count_default
-  computer_name_prefix            = null
-  sku                             = var.vm_size
-  tags                            = var.tags
-  zones                           = var.zones
-  zone_balance                    = var.zone_balance
-  provision_vm_agent              = false
+  name                 = var.name
+  computer_name_prefix = null
+  location             = var.location
+  resource_group_name  = var.resource_group_name
+
+  admin_username                  = var.authentication.username
+  admin_password                  = var.authentication.disable_password_authentication ? null : local.password
+  disable_password_authentication = var.authentication.disable_password_authentication
+
   dynamic "admin_ssh_key" {
-    for_each = var.ssh_keys
+    for_each = { for k, v in var.authentication.ssh_keys : k => v }
     content {
-      username   = var.username
-      public_key = ssh_keys.value
+      username   = var.authentication.username
+      public_key = admin_ssh_key.value
     }
   }
 
```

Comment:
As recently discussed, shall we add "allow_extension_operations" property with default set to false (how checkov likes it)? I created this variable in Panorama PR.